### PR TITLE
KEYCLOAK-7294: Password page - Angular

### PR DESCRIPTION
**Pull request by ssilvert**
Integrates with the new REST API for password change.

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java [hunk 1]
```diff
@@ -92,6 +93,7 @@ public void setLastUpdate(long lastUpdate) {
 
     }
 
+    @JsonIgnoreProperties(ignoreUnknown=true)
```
Comment (stianst) [inline services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java]: Why is this needed?

Comment (ssilvert) [reply services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java]: We need to ignore the password confirmation param as that will be checked on the client.  It's actually quite difficult to remove it from the request without writing lots of terrible, ugly, Angular code.  I didn't see any harm in just letting the REST API ignore it.

Comment (ssilvert) [reply services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java]: Password confirmation is the field in the UI that confirms the new password was typed in correctly.  You know, where you ask the user to retype his new password.

I'll change the REST side so that it does this check and throws an error if they don't match.  

The API won't complain if the confirmation is missing.
